pkg/asset/installconfig/azure/validation.go

```diff
@@ -319,3 +323,25 @@ func validateResourceGroup(client API, fieldPath *field.Path, platform *aztypes.
 	}
 	return allErrs
 }
+
+func validateAzureStackVHDURL(ARMEndpoint string, VHDURL string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if VHDURL == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("VHDURL"), "VHDURL must be set when installing on Azure Stack"))
+	} else {
+		storageParsedURL, err := url.Parse(ARMEndpoint)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("armEndpoint"), ARMEndpoint, "Failed to parse ARMEndpoint URL"))
+		}
+		storageBaseHost := strings.TrimPrefix(storageParsedURL.Host, "management.")
+		imageParsedURL, err := url.Parse(VHDURL)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("VHDURL"), VHDURL, "Failed to parse VHDURL URL"))
+		}
+		// If the URL for the image isn't in the Azure Stack environment we can't use it.
```

Comment:
If the URL is required to be part of the stack environment, is there benefit in querying azure to validate that the blob actually exists?

Reply:
Yes in the sense that we would fail earlier and more gracefully than a Terraform failure.

Reply:
I guess my question is do we have a sense of (1) how easy it is to go from the URL to querying the azure api for the blob and (2) how fragile that translation may be. The check for the existence would need to wait until the pre-terraform step, too.

Reply:
The blob URL should be predictable and easily translatable. But thinking about this more I wonder about permissions. I think it would be a valid use case to have a VHD in a blob in a separate subscription that the installer would not necessarily have permission to access via the api.

Reply:
What determines whether the VM is allowed to use the image if it is not the permissions used to create the VM? In other words, I would expect that if I have permission to create a VM using an image that I should also have permission to read that image. But, then, that is just my expectation and not a reflection on the reality of things.

Reply:
Nevertheless, we can certainly skip doing the extra validation that the blob exists. The error will come soon enough when the terraform runs.

Reply:
It is not permission to read the image, but rather the storage blob containing the VHD that will be used to create the image. In public Azure, Red Hat has a public storage blob `https://rhcos.blob.core.windows.net/imagebucket/rhcos-410.84.202112040202-0-azure.x86_64.vhd` that we copy into the installation target resource group, but I am doubting whether a user's credentials could use the Azure storage blob API to verify the existence of `https://rhcos.blob.core.windows.net/imagebucket/rhcos-410.84.202112040202-0-azure.x86_64.vhd`. I *think* there could be a similar valid use case in Azure Stack.

Also, this makes me wonder if the step of copying the VHD into the storage account is redundant for Azure Stack: https://github.com/openshift/installer/blob/master/data/data/azurestack/vnet/main.tf#L54-L67

I think we have the capability of testing both of these questions, but perhaps it would be better to scope those into a new PR.